### PR TITLE
cleaning up the email filters and adding slack filters

### PR DESCRIPTION
**Pull request by timatron**

### CODE REVIEW

**Review by mboynes — CHANGES_REQUESTED** [review]
[All hooks should have docblocks](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters)

File: class-sc-wp-updates-notifier.php [hunk 1]
```diff
@@ -448,12 +448,35 @@ public function check_themes_against_notified( $themes_need_update ) {
...
 			add_filter( 'wp_mail_content_type', array( $this, 'sc_wpun_wp_mail_content_type' ) ); // add content type filter
+		
+			/**
+			 * Filters the email content.
+			 *
+			 * Change the message that gets sent in the email.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $message Email message.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $message Email message.
```

File: class-sc-wp-updates-notifier.php [hunk 2]
```diff
@@ -448,12 +448,35 @@ public function check_themes_against_notified( $themes_need_update ) {
...
 			$settings = $this->get_set_options( self::OPT_FIELD ); // get settings
 	
+			/**
+			 * Filters the email subject.
+			 *
+			 * Change the subject line that gets sent in the email.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $subject Email subject line.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $subject Email subject line.
```

File: class-sc-wp-updates-notifier.php [hunk 3]
```diff
@@ -473,18 +496,75 @@ public function send_email_message( $message ) {
...
 			$settings = $this->get_set_options( self::OPT_FIELD ); // get settings
 
+			/**
+			 * Filters the Slack username.
+			 *
+			 * Change the username that is used to post to Slack.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $username Username string.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $username Username string.
```

File: class-sc-wp-updates-notifier.php [hunk 4]
```diff
@@ -473,18 +496,75 @@ public function send_email_message( $message ) {
...
+			$username = apply_filters( 'sc_wpun_slack_username', $username );
+			
+			/**
+			 * Filters the Slack user icon.
+			 *
+			 * Change the user icon that is posted to Slack.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $user_icon Emoji string.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $user_icon Emoji string.
```

File: class-sc-wp-updates-notifier.php [hunk 5]
```diff
@@ -473,18 +496,75 @@ public function send_email_message( $message ) {
...
+			$payload['channel'] = apply_filters( 'sc_wpun_slack_channel', $payload['channel'] );
+
+			/**
+			 * Filters the Slack webhook url.
+			 *
+			 * Change the webhook url that is called by the plugin to post to Slack.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $settings['slack_webhook_url'] Webhook url.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $settings['slack_webhook_url'] Webhook url.
```

File: class-sc-wp-updates-notifier.php [hunk 6]
```diff
@@ -473,18 +496,75 @@ public function send_email_message( $message ) {
...
 			}
 
+			/**
+			 * Filters the Slack channel.
+			 *
+			 * Change the Slack channel to post to.
+			 *
+			 * @since 1.6.1
+			 *
+			 * @param String  $payload['channel'] Slack channel.
```
Comment (renatonascalves) [inline class-sc-wp-updates-notifier.php]:
```suggestion
			 * @param string  $payload['channel'] Slack channel.
```

**Review by renatonascalves — APPROVED** [review]
🗡

**Pull request merged**